tests/test_model.py

```diff
@@ -332,3 +348,37 @@ def test_wrapper_meta_model():
 
     assert not torch.isnan(out).any()
     assert out.size() == big_features.size()
+
+
+def test_constrained_forecast():
```

Comment:
Nice test! I would have it do each type of constraint, as each of its own unit test.